iothub_client/tests/common_dt_e2e/iothubclient_common_dt_e2e.c

```diff
@@ -651,6 +668,48 @@ void dt_e2e_get_complete_desired_test(IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol,
 
     int initial_twin_version = dt_e2e_gettwin_version(serviceClientDeviceTwinHandle, deviceToUse);
 
+    // Receive first full twin before updating it.
+    time_t beginOperation, nowTime;
+    beginOperation = time(NULL);
+    while (
+        (nowTime = time(NULL)),
+        (difftime(nowTime, beginOperation) < MAX_CLOUD_TRAVEL_TIME) // time box
+        )
+    {
+        if (Lock(device->lock) != LOCK_OK)
+        {
+            ASSERT_FAIL("Lock failed");
+        }
+        else
+        {
+            if ((device->receivedCallBack) && (device->cb_payload != NULL))
+            {
+                LogInfo("device->cb_payload: %s", device->cb_payload);
+                int current_version = dt_e2e_parse_twin_version(device->cb_payload, false);
+
+                if (current_version != initial_twin_version)
+                {
+                    ASSERT_FAIL("The version of twin on callback (%d) did not match what was initially set (%d).",
+                                current_version, initial_twin_version);
+                }
+                Unlock(device->lock);
+                break;
+            }
+            Unlock(device->lock);
+        }
+        ThreadAPI_Sleep(1000);
+    }
+
+    ASSERT_IS_TRUE(difftime(nowTime, beginOperation) < MAX_CLOUD_TRAVEL_TIME, "Timeout waiting for twin message");
+
+    // Reset device for update.
+    device_desired_reset(device);
+
+    // Sleep still needed.  Device callback has been recorded in test app, but Hub is still completing twin PATCH subscription.
+    // This must be in place before the service SDK can update the twin, else the Hub will not send the PATCH
+    // message to the device and the desired twin callback will not be called.
+    ThreadAPI_Sleep(5000); // 5 seconds
```

Comment:
The `// 5 seconds` is redundant since it's already self-documented in the `5000` itself.